balancer/rls/internal/keys/builder_test.go

```diff
@@ -73,92 +114,70 @@ func TestMakeBuilderMap(t *testing.T) {
 		},
 		{
 			desc: "GrpcKeyBuilder with empty Service field",
 			cfg: &rlspb.RouteLookupConfig{
 				GrpcKeybuilders: []*rlspb.GrpcKeyBuilder{emptyServiceKeyBuilder, goodKeyBuilder1},
 			},
-			wantBuilderMap: nil,
-			wantErrPrefix:  "rls: GrpcKeyBuilder in RouteLookupConfig contains a Name field with no Service",
+			wantErrPrefix: "rls: GrpcKeyBuilder in RouteLookupConfig contains a Name field with no Service",
 		},
 		{
 			desc: "GrpcKeyBuilder with no Name",
 			cfg: &rlspb.RouteLookupConfig{
 				GrpcKeybuilders: []*rlspb.GrpcKeyBuilder{{}, goodKeyBuilder1},
 			},
-			wantBuilderMap: nil,
-			wantErrPrefix:  "rls: GrpcKeyBuilder in RouteLookupConfig does not contain any Name",
+			wantErrPrefix: "rls: GrpcKeyBuilder in RouteLookupConfig does not contain any Name",
 		},
 		{
 			desc: "GrpcKeyBuilder with requiredMatch field set",
 			cfg: &rlspb.RouteLookupConfig{
 				GrpcKeybuilders: []*rlspb.GrpcKeyBuilder{requiredMatchKeyBuilder, goodKeyBuilder1},
 			},
-			wantBuilderMap: nil,
-			wantErrPrefix:  "rls: GrpcKeyBuilder in RouteLookupConfig has required_match field set",
+			wantErrPrefix: "rls: GrpcKeyBuilder in RouteLookupConfig has required_match field set",
 		},
 		{
 			desc: "GrpcKeyBuilder two headers with same key",
 			cfg: &rlspb.RouteLookupConfig{
 				GrpcKeybuilders: []*rlspb.GrpcKeyBuilder{repeatedHeadersKeyBuilder, goodKeyBuilder1},
 			},
-			wantBuilderMap: nil,
-			wantErrPrefix:  "rls: GrpcKeyBuilder in RouteLookupConfig contains repeated Key field in headers",
+			wantErrPrefix: "rls: GrpcKeyBuilder in RouteLookupConfig contains repeated Key field in headers",
 		},
 		{
-			desc: "One good GrpcKeyBuilder",
+			desc: "GrpcKeyBuilder with slash in method name",
 			cfg: &rlspb.RouteLookupConfig{
-				GrpcKeybuilders: []*rlspb.GrpcKeyBuilder{goodKeyBuilder1},
+				GrpcKeybuilders: []*rlspb.GrpcKeyBuilder{methodNameWithSlashKeyBuilder},
 			},
-			wantBuilderMap: wantBuilderMap1,
-		},
-		{
-			desc: "Two good GrpcKeyBuilders",
-			cfg: &rlspb.RouteLookupConfig{
-				GrpcKeybuilders: []*rlspb.GrpcKeyBuilder{goodKeyBuilder1, goodKeyBuilder2},
-			},
-			wantBuilderMap: wantBuilderMap2,
+			wantErrPrefix: "rls: GrpcKeyBuilder in RouteLookupConfig contains a method with a slash",
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.desc, func(t *testing.T) {
 			builderMap, err := MakeBuilderMap(test.cfg)
-			if (err != nil) != (test.wantErrPrefix != "") {
-				t.Errorf("MakeBuilderMap(%+v) returned err: %v, want: %v", test.cfg, err, test.wantErrPrefix)
-			}
-			if (err != nil && !strings.HasPrefix(err.Error(), test.wantErrPrefix)) ||
-				!cmp.Equal(builderMap, test.wantBuilderMap, cmp.AllowUnexported(builder{}, matcher{})) {
-				t.Errorf("MakeBuilderMap(%+v) returned {%v, %v}, want: {%v, %v}", test.cfg, builderMap, err, test.wantBuilderMap, test.wantErrPrefix)
+			if builderMap != nil || err == nil || !strings.HasPrefix(err.Error(), test.wantErrPrefix) {
```

Comment:
One trick I've used in the past to avoid the special nil checks is:

```
if builderMap != nil || !strings.HasPrefix(fmt.Sprint(err), test.wantErrPrefix) {
```

Reply:
Done. Thanks.